### PR TITLE
Update library to Contributors library standards

### PR DESCRIPTION
**Pull request by maxdeviant**
This PR updates `purescript-arraybuffer-types` to adhere to the [Contributors library guidelines](https://github.com/purescript-contrib/governance/blob/main/library-guidelines.md).

This is part of the https://github.com/purescript-contrib/governance/issues/13 effort.

### CODE REVIEW

**Comment by maxdeviant** [comment]
Confirmed that `spago build` is working as expected:

```
λ spago build
[info] Installation complete.
[info] Build succeeded.
```

Once this PR gets merged I'll add issues for the following outstanding items:

- Adding a "quick start" guide
- Adding tests

File: README.md [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
+The quick start hasn't been written yet (contributions are welcome!). The quick start covers a common, minimal use case for the library, whereas longer examples and tutorials are kept in the [docs directory](./docs).
 
 ## Documentation
 
-Module documentation is [published on Pursuit](http://pursuit.purescript.org/packages/purescript-arraybuffer-types).
+`arraybuffer-types` documentation is stored in a few places:
+
+1. Module documentation is [published on Pursuit](https://pursuit.purescript.org/packages/purescript-arraybuffer-types).
+2. Written documentation is kept in the [docs directory](./docs).
+3. Usage examples can be found in [the test suite](./test).
```
Comment (thomashoneyman) [inline README.md]: We can remove parts of this section which aren't relevant for a library (for example, this one doesn't have tests and so we should remove that line. It probably doesn't need a `docs` directory, either).

Comment (thomashoneyman) [reply README.md]: Other than that, this looks good!

Comment (maxdeviant) [reply README.md]: Alright, I removed the `docs` directory and the references to `docs` and `test` in the README.

File: README.md [hunk 2]
```diff
@@ -25,8 +25,6 @@ The quick start hasn't been written yet (contributions are welcome!). The quick
 `arraybuffer-types` documentation is stored in a few places:
 
 1. Module documentation is [published on Pursuit](https://pursuit.purescript.org/packages/purescript-arraybuffer-types).
```
Comment (thomashoneyman) [inline README.md]: Given there is only one of these left, what if we just merge this with the prior sentence? Then there would be no numbered list.

“Module documentation for arraybuffer-types ...”

Comment (maxdeviant) [reply README.md]: I merged the sentences per your suggestion.

I also cleaned up another spot that was referring to the `docs` directory.

**Pull request merged**